soci/store/store.go

```diff
@@ -181,32 +200,32 @@ func (s *SociStore) BatchOpen(ctx context.Context) (context.Context, CleanupFunc
 }
 
 type ContainerdStore struct {
-	config.ContentStoreConfig
-	client *containerd.Client
+	ContentStoreConfig
 }
 
 // assert that ContainerdStore implements Store
 var _ Store = (*ContainerdStore)(nil)
 
-func NewContainerdStore(storeConfig config.ContentStoreConfig) (*ContainerdStore, error) {
-	client, err := containerd.New(storeConfig.ContainerdAddress)
-	if err != nil {
-		return nil, fmt.Errorf("could not connect to containerd socket for content store access: %w", err)
-	}
-
+func NewContainerdStore(storeConfig ContentStoreConfig) (*ContainerdStore, error) {
 	containerdStore := ContainerdStore{
-		client: client,
+		ContentStoreConfig: storeConfig,
 	}
 
-	containerdStore.ContentStoreConfig = storeConfig
+	if storeConfig.client == nil {
+		return nil, fmt.Errorf("containerd client is not provided; use WithClient to set it")
```

Comment:
It looks like this func is also used by soci cli, which won't pass in the client in. So we should provide a default implementation here - just for the CLI case.

Reply:
Actually we can just add WithClient to [this func](https://github.com/awslabs/soci-snapshotter/blob/main/cmd/soci/commands/internal/store.go#L27-L32)

Reply:
Done, think I did it right